ArchiSteamFarm/IPC/Startup.cs

```diff
@@ -211,17 +213,22 @@ internal sealed class Startup {
 
 					knownNetworks.Add(new IPNetwork(ipAddress, prefixLength));
 				}
+			} else {
+				// Use private address space networks by default, https://datatracker.ietf.org/doc/html/rfc1918#section-3
+				knownNetworks.Add(new IPNetwork(IPAddress.Parse("10.0.0.0"), 8));
+				knownNetworks.Add(new IPNetwork(IPAddress.Parse("172.16.0.0"), 12));
+				knownNetworks.Add(new IPNetwork(IPAddress.Parse("192.168.0.0"), 16));
 			}
 
+			KnownNetworks = knownNetworks.ToImmutableHashSet();
+
 			// Add support for proxies
 			services.Configure<ForwardedHeadersOptions>(
 				options => {
 					options.ForwardedHeaders = ForwardedHeaders.All;
 
-					if (knownNetworks != null) {
-						foreach (IPNetwork knownNetwork in knownNetworks) {
-							options.KnownNetworks.Add(knownNetwork);
-						}
+					foreach (IPNetwork knownNetwork in KnownNetworks) {
+						options.KnownNetworks.Add(knownNetwork);
```

Comment:
Earlier `X-Forwarded` headers by default were accepted only from loopback, and now they are accepted from any private network. Is this expected?

Reply:
Yes, I did it intentionally, I believe that it's a good default call to make, it's private subnet for a reason, and if somebody has an urge to explicitly stop trusting private networks, he can still do that by specifying `"KnownNetworks": [ ]` himself.

Reply:
I mean, if we trust in private addresses to skip a need of mandatory `IPCPassword`, we should also trust them specified in `KnownNetworks` for providing forwarded headers. It should be inter-connected imho, and I don't see a strong rationale for actually expecting from people to put their LANs in known networks, it's kinda known per-se. I believe it's correct to make it opt-out mechanism, unless you have some valid reasoning.

Reply:
I took into account some possible scenarios where this is unwanted, and I see basically only one: somebody using a VPN he does not trust which falls in one of the private networks.

Well, he shouldn't connect to that VPN to begin with, or set up firewall himself in this case. I very much doubt in somebody in the same VPN finding that ASF and attempting to brute-force it faking the connected IP addresses, for real.

It's a balance between convenience and security, that guy had to himself allow that local network to reach him by editing `IPC.config` from default `localhost`, he took that risk at that point, we don't want to spoon-feed the user treating him like an idiot, we should only limit to what we're 99.99% sure about, and we're 99.99% sure about people outside of private network range to be stopped from accessing ASF API unconditionally, within private network - we have to take a guess, and I'd rather guess that it's safe rather than not.

Reply:
It's back to what it was.